File: beekeeping-api/BeekeepingApi/Models/Harvest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace BeekeepingApi.Models
{
    public enum BeeProduct
    {
        BeeBread, Beeswax, Propolis, Honey
    }

    public class Harvest
    {
        [Key]
        public long Id { get; set; }

        [Required]
        public BeeProduct Product { get; set; }

        [Column(TypeName = "date")]
        [DisplayFormat(DataFormatString = "{yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        public DateTime StartDate { get; set; }

        [Column(TypeName = "date")]
        [DisplayFormat(DataFormatString = "{yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        public DateTime EndDate { get; set; }
        
        [Required]
        public double Quantity { get; set; }

        public long FarmId { get; set; }

        [ForeignKey("FarmId")]
        public Farm Farm { get; set; }

        public long? ApiaryId { get; set; }

        [ForeignKey("ApiaryId")]
        public Apiary Apiary { get; set; }

        public long? BeeFamilyId { get; set; }

        [ForeignKey("BeeFamilyId")]
        public BeeFamily BeeFamily { get; set; }
    }
}

File: beekeeping-app/src/app/_models/harvest.ts
export class Harvest {
    id: number;
    product: HarvestProducts;
    startDate: Date;
    endDate: Date;
    quantity: number;
    farmId: number;
    apiaryId: number;
    beeFamilyId: number;
    isDeleting: boolean = false;
}

export enum HarvestProducts {
    BeeBread,
    Beewax,
    Propolis,
    Honey
}

export const HarvestProduct2LabelMapping: Record<HarvestProducts, string> = {
    [HarvestProducts.BeeBread]: "Duonelė",
    [HarvestProducts.Beewax]: "Vaškas",
    [HarvestProducts.Propolis]: "Pikis",
    [HarvestProducts.Honey]: "Medus"
};
